docs/metasploit-framework.wiki/How-to-use-Railgun-for-Windows-post-exploitation.md

```diff
@@ -36,28 +36,30 @@ end
+There are four parameter/buffer directions: in, out, inout, and return. When you pass a value to an "in" parameter, Railgun handles the memory management. For example, `MessageBoxA` has an "in" parameter named `lpText`, and is of type PCHAR. You can simply pass a Ruby string to it, and Railgun handles the rest, it's all pretty straight forward.
 
-An "out" parameter will always be of a pointer datatype. Basically you tell Railgun how many bytes to allocate for the parameter, it allocates memory, provides a pointer to it when calling the function, and then it reads that region of memory that the function wrote, converts that to a Ruby object and adds it to the return hash.
+An "out" parameter will always be of a pointer datatype. Basically you tell Railgun how many bytes to allocate for the parameter, it allocates memory, provides a pointer to it when calling the function, and then it reads that region of memory that the function wrote, converts that to a Ruby object and adds it to the return hash. Some datatypes such as LPVOID and ULONG_PTR have a size that is determined based on the host architecture, e.g. 32-bit versions of Windows use 4-byte/32-bit values. For cross compatibility, the number 4 (for 4-bytes) can be used as the size for these values on both 32-bit and 64-bit systems. The number four comes from the size used for these types in the original 32-bit implementation and was selected to maintain backwards compatibility when 64-bit support was added.
 
 An "inout" parameter serves as an input to the called function, but can be potentially modified by it. You can inspect the return hash for the modified value like an "out" parameter.
 
-A quick way to define a new function at runtime can be done like the following example:
+The fourth type, "return" is used as the return data type. It is passed to `#add_function` after the function name argument.
+
+A quick way to define a new function (or redefine an existing function) at runtime can be done like the following example:
 
 ```ruby
 client.railgun.add_function('user32', 'MessageBoxA', 'DWORD',[
-	["DWORD","hWnd","in"],
-	["PCHAR","lpText","in"],
-	["PCHAR","lpCaption","in"],
-	["DWORD","uType","in"]
+	['DWORD','hWnd','in'],
+	['PCHAR','lpText','in'],
+	['PCHAR','lpCaption','in'],
+	['DWORD','uType','in']
  ])
 ```
 
-However, if this function will most likely be used more than once, or it's part of the Windows API, then you should put it in the library.
+However, if this function will most likely be used more than once, or it's part of the Windows API, then you should put it in to the library.
 
-### Usage
+## Usage
 
 The best way to try Railgun is with irb in a Windows Meterpreter prompt. Here's an example of how to get there:
 
```

Comment:
I can't comment on the line below, but if you add `msf` to the language type in the code snippet, like this:

https://github.com/rapid7/metasploit-framework/blame/c12ef82d35f8f595e6b1e5162b4246df0bb8bf21/docs/metasploit-framework.wiki/How-to-get-started-with-writing-a-post-module.md#L35

We'll get color highlighting when the site is generated 🎨